src/App.tsx

```diff
@@ -1,20 +1,40 @@
-import React from "react";
+import React, { useEffect} from "react";
 import { Routes, Route } from "react-router-dom";
 import Home from "./pages/home/Home";
 import Profile from "./pages/profile/Profile";
 import Feeds from "./pages/feed/Feeds";
-import Navbar from "./shared/navbar/Navbar";
+import { Signup } from "./pages/authentication/signup/Signup";
+import { Login } from "./pages/authentication/login/Login";
+import { PrivateRoute } from "./shared/privateRoute/PrivateRoute";
+import { ToastContainer } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import { getCurrentUser } from "./features/auth/authSlice";
+import { useAppDispatch } from "./app/hooks";
+
 
 function App() {
+
+  const dispatch = useAppDispatch();
+  useEffect(() => {
+    dispatch(getCurrentUser());
+  }, [dispatch]);
   return (
     <div className="App">
-      <Navbar />
+      <ToastContainer theme="colored" />
       <Routes>
-        <Route path="/" element={<Home />}>
+        <Route
+          path="/"
+          element={
+            <PrivateRoute path="/">
```

Comment:
Rename it as Require as per convention and it has some goal as in HOC

Reply:
??